File: eshop-backend/src/main/java/com/myapp/product/PagedResponse.java
package com.myapp.product;

import org.springframework.data.domain.Page;

import java.util.List;

public record PagedResponse<T>(
        List<T> content,
        int pageNumber,
        int pageSize,
        long totalElements,
        int totalPages,
        boolean last
) {
    public PagedResponse(List<T> content, Page<T> page) {
        this(content, page.getNumber(), page.getSize(), page.getTotalElements(), page.getTotalPages(), page.isLast());
    }
}

File: eshop-frontend/src/types/PagedResponse.ts
export interface PagedResponse<T> {
  content: T[];
  pageNumber: number;
  pageSize: number;
  totalElements: number;
  totalPages: number;
  last: boolean;
}

export interface RequestParam {
  page?: number; // default: 0
  size?: number; // default: 10
  sort?: string; // default: 'id,asc'
}